### PR TITLE
[docs]: Remove `api_spec.md`

### PR DESCRIPTION
**Pull request by 0x009922**
## Description

The PR with API Reference in the documentation is merged:

- https://github.com/hyperledger/iroha-2-docs/pull/417

This PR removes the original, obsolete `api_spec.md` and sanitizes all references into it.

I also made some tech writing chores.

### Linked issue

_None_

### Checklist

- [x] DevOps check
- [ ] Tech writers check of phrasing I used

### CODE REVIEW

**Review by WRRicht3r — DISMISSED** [review]

**Review by 6r1d — APPROVED** [review]

File: docs/source/iroha_2_whitepaper.md [hunk 1]
```diff
@@ -339,7 +339,7 @@ To retrieve information about World State View of the Peer clients will use Iroh
 
 ### 2.14 Client API
 
-Client libraries interact with Iroha over the HTTP and WebSocket. Check the up to date API reference [here](./references/api_spec.md).
+Client libraries interact with Iroha over the HTTP and WebSocket. Check the up-to-date [API Reference](https://hyperledger.github.io/iroha-2-docs/api/torii-endpoints).
```
Comment (yamkovoy) [inline docs/source/iroha_2_whitepaper.md:342]: I would change "up-to-date" for "latest", or omit it altogether.
Or we can aim to use the same structure for all of the similar sentences that direct a reader to other parts of the documentation.

In most cases, it is possible to use the following structure:
`<To learn more/For instructions> about/on <topic/term/action>, see [topic](link)`

Examples from the "Security" section:
- To learn more about generating cryptographic keys with `kagami`, available algorithms, and other parameters, see [Generating Cryptographic Keys with Kagami](../security/generating-cryptographic-keys.md#kagami).
- to learn more about cryptographic keys, see [Public Key Cryptography](./security/public-key-cryptography.md)

---

Here, for example, we can try the following:
"To learn more about the available endpoints and operations, see [API > Torii Endpoints](https://hyperledger.github.io/iroha-2-docs/api/torii-endpoints)

File: README.md [hunk 2]
```diff
@@ -155,17 +155,17 @@ A brief overview on how to configure and maintain an Iroha instance:
...
 
 You can provide configuration parameters either as a `config.json` or using environment variables. Refer to the [detailed list](./docs/source/references/config.md) of all available configuration parameters.
 
 Configuration example you may use as a reference point: [cli/src/samples.rs](./cli/src/samples.rs)
 
-## Endpoints
+### Endpoints
 
-You can find the detailed list of all available endpoints in the [API specifications](./docs/source/references/api_spec.md#endpoints).
+You can find the detailed list of all available endpoints in the [API Reference](https://hyperledger.github.io/iroha-2-docs/api/torii-endpoints).
```
Comment (yamkovoy) [inline README.md:166]: Try the following or something similar that you find fitting (i believe, it is a great idea to start off the sentence with stating the most important thing a user can find in the topic we're referencing):
`For a detailed list of all available endpoints and operations, see [API > Torii Endpoints](link-to-the-topic.md)`

A more detailed version:
`For a list of all endpoints, available operations, and ways to customize them with parameters, see [API > Torii Endpoints](link-to-the-topic.md)`

File: docs/source/iroha_2_whitepaper.md (hunk 1 above)
Comment (0x009922) [reply docs/source/iroha_2_whitepaper.md]: Applied

File: README.md (hunk 2 above)
Comment (0x009922) [reply README.md]: Done

**Review by 6r1d — APPROVED** [review]
Looks reasonable, approving

**Review by WRRicht3r — APPROVED** [review]

**Pull request merged**